awscli/customizations/generatecliskeleton.py

```diff
@@ -59,29 +65,66 @@ def _register_argument_action(self):
+            except IndexError:
+                pass
+            super(GenerateCliSkeletonArgument, self).override_required_args(
+                argument_table, args, **kwargs)
+
     def generate_json_skeleton(self, call_parameters, parsed_args,
                                parsed_globals, **kwargs):
-
-        # Only perform the method if the ``--generate-cli-skeleton`` was
-        # included in the command line.
-        if getattr(parsed_args, 'generate_cli_skeleton', False):
-
-            # Obtain the model of the operation
+        if getattr(parsed_args, 'generate_cli_skeleton', None):
+            for_output = parsed_args.generate_cli_skeleton == 'output'
             operation_model = self._operation_model
 
-            # Generate the skeleton based on the ``input_shape``.
-            argument_generator = ArgumentGenerator()
-            operation_input_shape = operation_model.input_shape
-            # If the ``input_shape`` is ``None``, generate an empty
-            # dictionary.
-            if operation_input_shape is None:
-                skeleton = {}
+            if for_output:
+                service_name = operation_model.service_model.service_name
+                operation_name = operation_model.name
+                return StubbedCLIOperationCaller(self._session).invoke(
+                    service_name, operation_name, call_parameters,
+                    parsed_globals)
             else:
-                skeleton = argument_generator.generate_skeleton(
-                    operation_input_shape)
-
-            # Write the generated skeleton to standard output.
-            sys.stdout.write(json.dumps(skeleton, indent=4))
-            sys.stdout.write('\n')
-            # This is the return code
-            return 0
+                argument_generator = ArgumentGenerator()
+                operation_input_shape = operation_model.input_shape
+                if operation_input_shape is None:
+                    skeleton = {}
+                else:
+                    skeleton = argument_generator.generate_skeleton(
+                        operation_input_shape)
+
+                sys.stdout.write(
+                    json.dumps(skeleton, indent=4, default=json_encoder)
+                )
+                sys.stdout.write('\n')
+                return 0
+
+
+class StubbedCLIOperationCaller(CLIOperationCaller):
+    """A stubbed CLIOperationCaller
+
+    It generates a fake response and uses the response and provided parameters
+    to make a stubbed client call for an operation command.
+    """
+    def _make_client_call(self, client, operation_name, parameters,
+                          parsed_globals):
+        method_name = xform_name(operation_name)
+        operation_model = client.meta.service_model.operation_model(
+            operation_name)
+        fake_response = {}
+        if operation_model.output_shape:
+            argument_generator = ArgumentGenerator(use_member_names=True)
```

Comment:
I think we may need to update the logic in botocore here, especially if we're going through the param validation.  For example:

```
$ aws dynamodb list-tables --generate-cli-skeleton output

Parameter validation failed:
Invalid length for parameter TableNames[0], value: 0, valid range: 3-inf
```